File: src/Dfe.PlanTech.Domain/Questionnaire/Models/Question.cs
using Dfe.PlanTech.Domain.Content.Models;
using Dfe.PlanTech.Domain.Questionnaire.Interfaces;

namespace Dfe.PlanTech.Domain.Questionnaire.Models;

public class Question : ContentComponent, IQuestion<Answer>
{
    /// <summary>
    /// Actual question text
    /// </summary>
    public string? Text { get; init; }

    /// <summary>
    /// Optional help text
    /// </summary>
    public string? HelpText { get; init; }

    public List<Answer> Answers { get; init; } = new();

    public string? Slug { get; set; }
}

File: contentful/export-processor/content-types/question.js
import { Answer } from '#src/content-types/answer';

export class Question {
  answers;
  text;
  helpText;
  slug;
  id;

  constructor({ fields, sys }) {
    this.answers = fields.answers.map((answer) => new Answer(answer));
    this.text = fields.text;
    this.helpText = fields.helpText;
    this.slug = fields.slug;
    this.id = sys.id;
  }

  get answerIds() {
    return this.answers.map((answer) => answer.id);
  }
}
